server/backend/testspg/crud/test_crud_application.py

```diff
@@ -1,34 +1,167 @@
-import os
-import sys
-from sqlalchemy.orm import sessionmaker
+from sqlalchemy.orm import Session
+import logging
+import datetime
+from api.app.crud import crud_application
+import api.app.schemas as schemas
+import api.app.constants as constants
+from testspg.constants import TEST_NOT_EXIST_APPLICATION_ID
 
-sys.path.append(os.path.join(os.path.dirname(__file__), "../.."))
+LOGGER = logging.getLogger(__name__)
 
-from api.app.crud import crud_application
+TEST_APPLICATION_NAME_NOT_FOUND = "NOT_FOUND"
+
+TEST_APPLICATION_NAME_FAM = "FAM"
+TEST_APPLICATION_ID_FAM = 1
+TEST_APPLICATION_ROLE_ID_FAM = 1
+
+TEST_APPLICATION_NAME_FOM_DEV = "FOM_DEV"
+TEST_APPLICATION_ID_FOM_DEV = 2
+TEST_APPLICATION_ROLES_FOM_DEV = ["FOM_SUBMITTER", "FOM_REVIEWER"]
+TEST_APPLICATION_ROLE_ID_FOM_DEV = 3
 
+NEW_APPLICATION = {
+    "application_name": "TEST_APP_DEV",
+    "application_description": "A testing application",
+    "app_environment": constants.AppEnv.APP_ENV_TYPE_DEV,
+    "create_user": constants.FAM_PROXY_API_USER,
+    "create_date": datetime.datetime.now(),
+    "update_user": "Tester",
+    "update_date": datetime.datetime.now(),
+}
 
-def test_get_applications(dbPgSession: sessionmaker):
-    apps = crud_application.get_applications(db=dbPgSession)
+
+def test_get_applications(db_pg_connection: Session):
+    apps = crud_application.get_applications(db=db_pg_connection)
     assert len(apps) == 7
```

Comment:
I wonder if this "7" is too magical to put into test... and will fail each time onboarding new app.

Reply:
Yes, maybe I should change it to "> 1"?

Reply:
Updated to > 1

Reply:
> I think it should be `assert len(apps) == 42` because 42 is the answer to all questions! https://en.wikipedia.org/wiki/42_(number)#:~:text=The%20number%2042%20is%2C%20in,period%20of%207.5%20million%20years.

Ah~ Nice to see you master Fran; didn't know you are also a mathmatician! Thanks for the number~ 42 will be forever in my lotto ticket from now on!